src/rules/noDuplicateLetRule.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "duplicate block-scoped variable: '";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoDuplicateLetWalker(sourceFile, this.getOptions()));
+    }
+}
+
+class NoDuplicateLetWalker extends Lint.BlockScopeAwareRuleWalker<ScopeInfo> {
+    public createScope(): ScopeInfo {
+        return new ScopeInfo();
+    }
+
+    public visitVariableDeclaration(node: ts.VariableDeclaration): void {
```

Comment:
This is going to hit the same issue that the noDuplicateVariableRule hit, which is that the "variable declaration" in catch blocks seems to have its own special rules. You should probably try to skip that catch-block variable declaration.